### ISSUE TITLE
Upgrade Pluggy depdendency version (<1.4) - Preventing upgrade of Pytest 8.1 that requires pluggy >=1.4

### ISSUE THREAD

**Issue by rxm7706** [kedro-org/kedro#3763]
## Description
Kedro currently pins Pluggy to >=1.0,<1.4
https://github.com/kedro-org/kedro/pull/3574
which was pinned because it broke tests
https://github.com/kedro-org/kedro/pull/3552

This is currently preventing upgrade of pytest - as pytest 8.1 requires pluggy >=1.4,<2.0

## Actual Result

```
-- If you received an error, place it here.
```

```
-- Separate them if you have more than one.
```

## Your Environment

* Kedro version used (`pip show kedro` or `kedro -V`):
* Python version used (`python -V`):
* Operating system and version:

**Comment by noklam** [comment]
Created a draft PR to see what breaks pluggy. Curious if there is any goodies for pytest 8.0? We are still on <8.0.

**Comment by antonymilne** [comment]
Just to add another data point here. I was also tripped up by this when running vizro tests which use pytest and have kedro as a dependency.

**Comment by noklam** [comment]
@antonymilne the dependency is removed already. I think we may have accidentally pin pytest as core dependencies in some version, it should be test dependencies only.

**Comment by antonymilne** [comment]
@noklam the problem is with the kedro's dependence on`pluggy` than on `pytest`. But it seems to be fixed here 🙂 https://github.com/kedro-org/kedro/pull/3840

**Comment by noklam** [comment]
@antonymilne Sorry I mean to say pluggy but yeah the pin was removed.

Before the new pluggy release it was necessary to pin it as test dependencies as it breaks certain API that we mock

**Comment by noklam** [comment]
Fixed by #3840 already
